clis/teliod/src/daemon.rs

```diff
@@ -47,32 +65,112 @@ fn telio_task(
+        Box::new(MockSystemConfigurator::new(false))
+    } else {
+        Box::new(IfconfigSystemConfigurator)
+    };
+
     // TODO: This is temporary to be removed later on when we have proper integration
     // tests with core API. This is to not look for tokens in a test environment
     // right now as the values are dummy and program will not run as it expects
     // real tokens.
     if !auth_token.as_str().eq("") {
         start_telio(&mut telio, node_identity.private_key, interface_name)?;
-        task_retrieve_meshmap(node_identity, auth_token, tx_channel);
+
+        // Auto enable meshnet for the first time
+        if auto_start_meshnet {
+            task_enable_meshnet(tx_channel.clone());
+        }
 
         while let Some(cmd) = rx_channel.blocking_recv() {
             info!("Got command {:?}", cmd);
             match cmd {
                 TelioTaskCmd::UpdateMeshmap(map) => {
-                    if let Err(e) = telio.set_config(&Some(map)) {
-                        error!("Unable to set meshmap due to {e}");
+                    let new_ip_address = map
+                        .ip_addresses
+                        .as_deref()
+                        .and_then(|addresses| addresses.first().cloned());
+                    let address_changed = state.interface_ip_address != new_ip_address;
+                    state.interface_ip_address = new_ip_address;
+
+                    if state.meshnet_is_enabled {
+                        match telio.set_config(&Some(map)) {
+                            Ok(_) => {
+                                // configure the interface ip address
+                                if address_changed && auto_configure_interface {
+                                    task_set_ip(tx_channel.clone());
+                                }
+                            }
+                            Err(e) => {
+                                error!("Unable to set meshmap due to {e}");
+                            }
+                        }
                     }
                 }
                 TelioTaskCmd::GetStatus(response_tx_channel) => {
                     let status_report = TelioStatusReport {
-                        is_running: telio.is_running(),
-                        nodes: telio.external_nodes()?,
+                        telio_is_running: telio.is_running(),
+                        meshnet_enabled: state.meshnet_is_enabled,
+                        meshnet_ip: state.interface_ip_address,
+                        external_nodes: telio.external_nodes()?,
                     };
                     debug!("Telio status: {:#?}", status_report);
                     if response_tx_channel.send(status_report).is_err() {
                         error!("Telio task failed sending status report")
                     }
                 }
+                TelioTaskCmd::SetMeshnet(enabled, response_tx_channel) => {
```

Comment:
Why are we setting/unsetting the meshmap? Shouldn't this command kill or start telio?

Reply:
Let's discuss this further under this [thread](https://github.com/NordSecurity/libtelio/pull/953#discussion_r1853572042)